src/output.c

```diff
@@ -49,111 +33,141 @@ static char *malloc_sprintf(const char *fmt, ...)
+    else
+        linker_flags = strdup("-lm");
+
+    List(char) quoted_object_files = {0};
+    for (int i = 0; objpaths[i]; i++) {
+        if (i)
+            Append(&quoted_object_files, ' ');
+        Append(&quoted_object_files, '"');
+        AppendStr(&quoted_object_files, objpaths[i]);
+        Append(&quoted_object_files, '"');
+    }
+    Append(&quoted_object_files, '\0');
 
     char *command;
 #ifdef _WIN32
     // Assume mingw with clang has been downloaded with windows_setup.sh.
     // Could also use clang, but gcc has less dependencies so we can make the Windows zips smaller.
     // Windows quoting is weird. The outermost quotes get stripped here.
-    command = malloc_sprintf("\"\"%s\\mingw64\\bin\\gcc.exe\" \"%s\" -o \"%s\" %s\"", instdir, objpath, exepath, linker_flags);
+    command = malloc_sprintf("\"\"%s\\mingw64\\bin\\gcc.exe\" %s -o \"%s\" %s\"", instdir, quoted_object_files.ptr, exepath, linker_flags);
 #else
     // Assume clang is installed and use it to link. Could use lld, but clang is needed anyway.
     (void)instdir;
-    command = malloc_sprintf("'%s' '%s' -o '%s' %s", JOU_CLANG_PATH, objpath, exepath, linker_flags);
+    command = malloc_sprintf("'%s' %s -o '%s' %s", JOU_CLANG_PATH, quoted_object_files.ptr, exepath, linker_flags);
 #endif
+    free(quoted_object_files.ptr);
+    free(jou_exe);
+    free(linker_flags);
 
     if (command_line_args.verbosity >= 2)
         printf("Running linker: %s\n", command);
     else if (command_line_args.verbosity >= 1)
         printf("Running linker\n");
+
     if (system(command))
         exit(1);
-
-    free(jou_exe);
     free(command);
-    free(linker_flags);
 }
 
-static char *get_filename_without_suffix(const LLVMModuleRef module)
+static void mkdir_exist_ok(const char *p)
 {
-    const char *filename = LLVMGetSourceFileName(module, (size_t[]){0});
-
-    if (strrchr(filename, '/'))
-        filename = strrchr(filename, '/') + 1;
-#ifdef _WIN32
-    if (strrchr(filename, '\\'))
-        filename = strrchr(filename, '\\') + 1;
-#endif
-
-    int len = strlen(filename);
-    if (len>4 && !strcmp(&filename[len-4], ".jou"))
-        len -= 4;
-
-    char *result = malloc(len+1);
-    memcpy(result, filename, len);
-    result[len] = '\0';
-
-    return result;
+    if (jou_mkdir(p) == 0 || errno == EEXIST)
+        return;
+    fprintf(stderr, "%s: cannot create directory \"%s\": %s\n", command_line_args.argv0, p, strerror(errno));
+    exit(1);
 }
 
-char *get_path_to_file_in_jou_compiled(LLVMModuleRef module, const char *filename)
+static char *get_path_to_file_in_jou_compiled(const char *filename)
 {
-    char *sourcepath = strdup(LLVMGetSourceFileName(module, (size_t[]){0}));
-
-    char *result = malloc(strlen(sourcepath) + strlen(filename) + 100);
-    char *tmp = strdup(sourcepath);
-    sprintf(result, "%s/jou_compiled", dirname(tmp));
-    free(tmp);
+    // Place it to current working directory. We need subdirectories based on the
```

Comment:
Maybe should instead be directory containing main Jou file, so you don't get race condition if multiple directory contain same name file